### ISSUE TITLE
Does not work on OTP 18 with elixir 1.0.5 - 1.2.0 

### ISSUE THREAD

**Issue by dm1try** [lexmag/msgpax#7]
no luck on 1.2.0 and 10 failures on 1.1.1 (all the same `Msgpax.Packer not implemented for...`)

```elixir
[msgpax 1.2.0 master] mix test   
** (CompileError) test/msgpax/ext_test.exs:39: undefined function packed/0
    (stdlib) lists.erl:1337: :lists.foreach/2
    (stdlib) erl_eval.erl:669: :erl_eval.do_apply/6

[msgpax 1.2.0 master] kiex use 1.1.1
Using 1.1.1
[msgpax 1.1.1 master] mix clean
[msgpax 1.1.1 master] mix test
Compiled lib/msgpax.ex
Compiled lib/msgpax/bin.ex
Compiled lib/msgpax/ext.ex
Compiled lib/msgpax/unpacker.ex
Compiled lib/msgpax/packer.ex
Generated msgpax app
.

  1) test empty options (Msgpax.ExtTest)
     test/msgpax/ext_test.exs:82
     ** (Protocol.UndefinedError) protocol Msgpax.Packer not implemented for %Msgpax.ExtTest.Sample{seed: "A", size: 1}
     stacktrace:
       (msgpax) lib/msgpax/packer.ex:14: Msgpax.Packer.impl_for!/1
       (msgpax) lib/msgpax/packer.ex:15: Msgpax.Packer.transform/1
       (msgpax) lib/msgpax/packer.ex:18: Msgpax.Packer.pack/1
       test/msgpax/ext_test.exs:85
...

............................

Finished in 1.1 seconds (0.7s on load, 0.3s on tests)
45 tests, 10 failures

Randomized with seed 771999
[msgpax 1.1.1 master]
```

Erlang/OTP 18 [erts-7.2] [source-e6dd627] [64-bit]  Ubuntu

**Comment by lexmag** [comment]
Great catch, thank you!
Indeed, there is a `CompileError` error on Elixir v1.2.0, but all the other versions are good.
You're getting an `Protocol.UndefinedError` error on v1.1 because protocols was consolidated during previous run on v1.2.
I've just pushed a commit that fixes this behaviour 19046479b7e26eb94be29ba89dc606a20c98730b, and now I'm looking into a failure on v1.2.

**Comment by dm1try** [comment]
@lexmag thanks, it works for now :)

**Comment by lexmag** [comment]
@dm1try regarding compilation failure on v1.2, it is a regression in Elixir (please see https://github.com/elixir-lang/elixir/issues/4174).

**Comment by dm1try** [comment]
@lexmag, oh ok.
I forgot to mention that I changed the assertions a little bit:
```diff
+      {:ok, packed} = Msgpax.pack(unquote(input))
+      <<unquote_splicing(format), _::bytes>> = IO.iodata_to_binary(packed)
+      {:ok, unpacked} = Msgpax.unpack(packed, unquote(opts))
+      assert(unpacked == unquote(output))
```
so this is why it works for me :) indeed, it's the regression

**Comment by lexmag** [comment]
Although dropping `assert` will make it to compile, sadly, we will lose very helpful output on failure. :cry:

**Comment by dm1try** [comment]
agree, my fault (I was panicked :innocent:)

**Comment by dm1try** [comment]
Does it worth to mention in `README` that you should disable `protocol_consolidation` on the test environment for your app if you create own protocols in your tests?

**Comment by dm1try** [comment]
@lexmag the last question related to this issue https://github.com/elixir-lang/elixir/issues/4175

**Comment by lexmag** [comment]
I think regardless of elixir-lang/elixir#4175, it worth to add `protocol_consolidation` information in the `README`.

**Comment by lexmag** [comment]
Hey @dm1try, the fix has just landed in Elixir: https://github.com/elixir-lang/elixir/commit/a78e0513e96ce36fb8e58fa1e13d48658a0ffa2f.

**Comment by dm1try** [comment]
> Hey @dm1try, the fix has just landed in Elixir: elixir-lang/elixir@a78e051.

@lexmag great!
